algorithm-prototyping/brute-force-prototype/src/main/java/edu/uga/devdogs/bruteforceprototype/BruteForcePrototype.java

```
@@ -11,6 +11,33 @@
 
 public class BruteForcePrototype {
 
+
+    /**
+     * Main method for the BruteForcePrototype class.
+     * Calls most of the other functions, manages error handling, and handles I/O.
+     *
+     * @param inputCourses a set of requested courses to generate an optimal schedule from
+     * @param distances a nested string map that represents distances between buildings on campus
+     * @param weights an array of floats representing the weights for each objective
+     * @param constraints the soft constraints on the schedule
+     * @return the output of optimize()
+     */
+    public static int[][] algorithmDriver(Set<Course> inputCourses, Map<String, Map<String, Double>> distances,  double[] weights, SConstraints constraints){
+        Set<Course> outputCourses;
+
+        try{
```

Lines 28 through 38 need to be changed. I probably should've clarified some things in the issue description. What eventually will happen is if `dataPreFilter` has no course that passes through it, instead a overloaded version of our `optimize` algorithm will run. The overloaded `optimize` has not been implemented yet, but will begin implementation next week. Do not implement it yet, or worry about calling a new version. Simply call `optimize` on both branches of the error handling. In addition, `dataPreFilter` will need to change. I once again apologize for not realizing that on your last pull request. Make it to where if no classes make it past the filter, it should throw an exception which is caught here. Let me know if further clarification is need!